### PR TITLE
Touch Switch Implementation Issue 144

### PR DESCRIPTION
**Pull request by joefol**
Implementing touch switch helper and controller.

### CODE REVIEW

File: components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import io.micronaut.http.annotation.Get;
+
+@Controller("/touchSwitch")
+public class TouchSwitchController {
+
+    private final TouchSwitchHelper touchSwitchHelper;
+
+    private final LEDHelper ledHelper;
+
+    public TouchSwitchController(DigitalInput touchSwitch,
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java:17]: Use @Named to fetch configuration from Application.yml file for touchSwitch and led

File: components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                                 DigitalOutput led) {
+        this.touchSwitchHelper = new TouchSwitchHelper(touchSwitch);
+        this.ledHelper = new LEDHelper(led);
+    }
+
+    @Get("/enable")
+    public void initController(){
+        touchSwitchHelper.addEventListener(e->{
+            if(touchSwitchHelper.isTouched){
+                ledHelper.switchState();
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java:27]: Use ledOn() and ledOff() here. For some reason if the led turned on in the initial state of the application, switching state actions might get swapped. One more approach is to set led off initially via constructor

File: components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            }
+            else if(touchSwitchHelper.isReleased){
+                ledHelper.switchState();
+            }
+        });
+    }
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java:33]: add disable function and make use of removeEventListener from helper

**Review by ruthvikm — CHANGES_REQUESTED** [review]

File: components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java [hunk 4]
```diff
@@ -32,4 +32,16 @@ else if(touchSwitchHelper.isReleased){
             }
         });
     }
+
+    @Get("/disable")
+    public void disableController() {
+        touchSwitchHelper.removeEventListener(e->{
+            if(touchSwitchHelper.isTouched){
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/TouchSwitchController.java:39]: I don't think we need this if else block here

**Review by ruthvikm — CHANGES_REQUESTED** [review]

**Review by ruthvikm — COMMENTED** [review]
Add comments which are appropriate for javadocs

**Review by ruthvikm — DISMISSED** [review]

**Review by SwindleA — APPROVED** [review]

**Review by adelgadoj3 — APPROVED** [review]

**Review by ruthvikm — APPROVED** [review]

**Pull request merged**